query-parser/src/parse.rs

```diff
@@ -0,0 +1,272 @@
+// Copyright 2016 Mozilla
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+#![allow(dead_code)]
+#![allow(unused_imports)]
+#![allow(unused_variables)]
+
+extern crate combine;
+extern crate edn;
+extern crate mentat_query;
+
+use std::collections::BTreeMap;
+
+use self::combine::{any, eof, many1, optional, parser, satisfy_map, token, Parser, ParseResult,
+                    Stream};
+use self::combine::combinator::{Expected, FnParser, choice, try};
+use self::edn::Value::PlainSymbol;
+use self::mentat_query::{Element, FindSpec, SrcVar, Variable};
+
+use super::error::{FindParseError, FindParseResult};
+
+pub struct FindSp<I>(::std::marker::PhantomData<fn(I) -> I>);
```

Comment:
Can we s/FindSp/FindSpec/?  I guess not due to namespacing?  Maybe just a comment that `FindSp` stands for something.